### PR TITLE
Pretrained model tests

### PR DESCRIPTION
**Pull request by RaunoArike**
- Tests the functionality in `src/models/pretrained/cifar_10.jl`, `src/models/pretrained/mnist.jl`, and `src/models/pretrained/fashion_mnist.jl`

### CODE REVIEW

File: test/models/pretrained.jl [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+                @testset "$name" begin
+                    counterfactual_data = Data.data_catalogue[:vision][key]()
+                    X = counterfactual_data.X
+                    # Randomly selected factual:
+                    Random.seed!(123)
+                    x = select_factual(counterfactual_data, rand(1:size(X, 2)))
+                    # Choose target:
+                    y = predict_label(M, counterfactual_data, x)
+                    target = get_target(counterfactual_data, y[1])
+                    generator = GravitationalGenerator()
```
Comment (A-Salmon) [inline test/models/pretrained.jl:31]: Is there a reason why you chose the gravitational generator here?

**Review by A-Salmon — COMMENTED** [review]
Solid job, tests look really good. There's one thing I had a question about, once this is addressed I'll approve.

File: test/models/pretrained.jl (hunk 1 above)
Comment (RaunoArike) [reply test/models/pretrained.jl:32]: No particular reason - it just takes too much time to iterate through all generators for all models and gravitational seemed like a good representative of generators that don't require latent space and can be used for both binary and multi-class classification problems. Added a clarifying comment to the test.

**Review by VincentPikand — APPROVED** [review]
lgtm

**Review by A-Salmon — APPROVED** [review]

**Pull request merged**